app/graphql/client.tsx

```diff
@@ -9,10 +9,34 @@ export const client = createClient({
   url: GRAPHQL_ENDPOINT,
   exchanges: [...devtoolsExchanges, ...defaultExchanges],
   fetchOptions: {
-    headers: {
-      "x-visualize-cache-control": flag("server-side-cache.disable")
-        ? "no-cache"
-        : "",
-    },
+    headers: { ...getCacheControlHeader(), ...getDebugModeHeader() },
   },
 });
+
+function getCacheControlHeader() {
+  const disableCache = flag("server-side-cache.disable");
+
+  if (disableCache) {
+    return {
+      "x-visualize-cache-control": "no-cache",
+    };
+  } else {
+    return {
+      "x-visualize-cache-control": "",
+    };
+  }
+}
+
+function getDebugModeHeader() {
+  const debugMode = flag("debug");
+
+  if (debugMode) {
+    return {
+      "x-visualize-debug": "true",
+    };
+  } else {
+    return {
+      "x-visualize-debug": "",
+    };
+  }
+}
```

Comment:
Nit: Would find it simpler to have only one function getHeaders doing both things.

Reply:
Thanks, done!